lib/src/rules/prefer_final_over_var.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.prefer_final_over_var;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/linter.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc =
+    r'Prefer final for variable declaration if reference is not reassigned.';
+
+const _details = r'''
+
+**DO** Prefer declaring variables as final if they are not reassigned later in the code.
+
+**BAD:**
+```
+void badMethod() {
+  var label = 'hola mundo!'; // LINT
+  print(label);
+}
+```
+
+**BAD:**
+```
+class BadImmutable {
+  var label = 'hola mundo!'; // LINT
+  ...
+}
+```
+
+**GOOD:**
+```
+class GoodMutable {
+  var label = 'hola mundo!';
+
+  void changeLabel() {
+    label = 'hello world!';
+  }
+}
+```
+
+**GOOD:**
+```
+class GoodMutable {
+  var label = 'hola mundo!';
+
+  void changeLabel() {
+    label = 'hello world!';
+  }
+}
+```
+
+**GOOD:**
+```
+void goodMethod() {
+  final label = 'hola mundo!';
+  print(label);
+}
+```
+
+**GOOD:**
+```
+void mutableCase() {
+  var label = 'hola mundo!';
+  print(label);
+  label = 'hello world';
+  print(label);
+}
+```
+''';
+
+class PreferFinalOverVar extends LintRule {
+  _Visitor _visitor;
+  PreferFinalOverVar()
+      : super(
+            name: 'prefer_final_over_var',
+            description: _desc,
+            details: _details,
+            group: Group.style) {
+    _visitor = new _Visitor(this);
+  }
+
+  @override
+  AstVisitor getVisitor() => _visitor;
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+  _Visitor(this.rule);
+
+  @override
+  void visitVariableDeclaration(VariableDeclaration node) {
+    if (node.isConst || node.isFinal) {
+      return;
+    }
+
+    FunctionBody function = node.getAncestor((a) => a is FunctionBody);
+    if (function != null &&
+        !function.isPotentiallyMutatedInScope(node.element)) {
+      rule.reportLint(node);
+    }
+
+    ClassDeclaration classDeclaration =
```

Comment:
Separated in two rules and added a third one for function and method parameters.